rfd/0083-license-expiration-warnings.md

```diff
@@ -0,0 +1,135 @@
+```
+
+```
+$ build/tsh login --proxy=localhost:9876 --user someUser
+Profile URL:        https://localhost:9876
+Logged in as:       someUser
+Cluster:            localhost
+Roles:              access, editor
+Logins:             someUser
+Valid until:        2022-08-16 03:38:07 +0100 IST [valid for 12h0m0s]
+Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty
+
+Your Teleport Enterprise Edition license has expired. Please reach out to [licenses@goteleport.com](mailto:licenses@goteleport.com) to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+```
+$ build/tctl status -c /home/edward/teleport/teleport.yaml
+Cluster  localhost                                                               
+Version  10.1.2                                                                  
+host CA  never updated                                                           
+user CA  never updated                                                           
+db CA    never updated                                                           
+jwt CA   never updated                                                           
+CA pin   sha256:3d72102f020146d09ff400810f59f70b8163ebfe6ec1ecfb0b3b2a0c151592
+
+Your Teleport Enterprise Edition license has expired. Please reach out to [licenses@goteleport.com](mailto:licenses@goteleport.com) to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+
+CLI does not support snoozing the warnings as you can in the webUI. Which provides the functionality to disable the warnings temporarily.
+
+### Web UI UX
+
+Web UI will display license expiration banner on top of the page with the following rules:
+Show warning (yellow) to users N days prior to expiration, 1 day snooze, based on license_warning_days role option with same warning text as the CLI
+Show error (red) to all users after expiration, no dismiss or snoozing available.
+Snoozing allows users to disable the warning in the web UI from being displayed for N days.
+
+### Configuration UX
+
+The time interval in which to show a warning before expiry can be configured on a per role basis.
+Interval can be configured by modifying a role option ‘license_warning_days’.
+Default will be 90 days, admins can configure if needed.
+Default preset roles will use the default. For example the editor/auditor/access roles.
+When multiple roles for the same user have different values for this interval the longest one will be used.
+This value cannot be configured to be below the minimum value of 10 days.
+
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: editor
+spec:
+  options:
+    license_warning_days: 90  # ← defaults to 30, min 10
```

Comment:
@EdwardDowling Let's remove configurable time interval from scope as discussed with @xinding33.

Reply:
Sounds good to me, this should simplify everything.